### PR TITLE
add missing configuration values to telemetry

### PR DESCRIPTION
**Pull request by ida613**
### What does this PR do?
Send all configuration values to telemetry in event app-started, except for the ones on the blocklist. 

### Plugin Checklist

- [ ] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

File: packages/dd-trace/src/config.js [hunk 1]
```diff
@@ -478,9 +398,13 @@ class Config {
 
     const defaults = setHiddenProperty(this, '_defaults', {})
 
+    this._setValue(defaults, 'appsec.apiSecurity.enabled', true)
+    this._setValue(defaults, 'appsec.apiSecurity.requestSampling', 0.1)
+    this._setValue(defaults, 'appsec.blockedTemplateGraphql', undefined)
     this._setValue(defaults, 'appsec.blockedTemplateHtml', undefined)
     this._setValue(defaults, 'appsec.blockedTemplateJson', undefined)
     this._setValue(defaults, 'appsec.enabled', undefined)
+    this._setValue(defaults, 'appsec.eventTracking', { enabled: true, mode: 'safe' })
```
Comment (simon-id) [inline packages/dd-trace/src/config.js:407]: shouldn't that be split in different props instead ???
```suggestion
    this._setValue(defaults, 'appsec.eventTracking.enabled', true)
    this._setValue(defaults, 'appsec.eventTracking.mode', 'safe')
```
and also down below ?

**Review by simon-id — CHANGES_REQUESTED** [review]

File: packages/dd-trace/src/config.js (hunk 1 above)
Comment (ida613) [reply packages/dd-trace/src/config.js:407]: there is no [mapping](https://github.com/DataDog/dd-go/blob/prod/trace/apps/tracer-telemetry-intake/telemetry-payload/static/config_norm_rules.json) for appsec.eventTracking.enabled or appsec.eventTracking.mode. This means when other languages send this configuration through telemetry to the backend, they send appsec.eventTracking directly.

Comment (simon-id) [reply packages/dd-trace/src/config.js:407]: Im going to ask the other teams to see if we should add the mapping maybe

Comment (uurien) [reply packages/dd-trace/src/config.js:407]: Hi Ida,
`enabled` field of `appsec.eventTracking` is just for internal use, it is inferred from `mode` [here](https://github.com/DataDog/dd-trace-js/blob/367c693c2e87a6314bfa139785a28e08c1f52342/packages/dd-trace/src/config.js#L369)

I think that we should send only `appsec.eventTracking.mode` and add this key in mappings. It is the reason why the other libraries are sending only one field and not two.

Comment (ida613) [reply packages/dd-trace/src/config.js:407]: thank you @uurien I'll make the changes

Comment (ida613) [reply packages/dd-trace/src/config.js]: the changes have been made :)

File: packages/dd-trace/src/config.js [hunk 2]
```diff
@@ -810,11 +791,20 @@ class Config {
 
     tagger.add(tags, options.tags)
 
+    this._setBoolean(opts, 'appsec.apiSecurity.enabled', options.appsec?.apiSecurity?.enabled)
+    this._setUnit(opts, 'appsec.apiSecurity.requestSampling', options.appsec?.apiSecurity?.requestSampling)
```
Comment (simon-id) [inline packages/dd-trace/src/config.js:795]: superfluous optional chaining
```suggestion
    this._setBoolean(opts, 'appsec.apiSecurity.enabled', options.appsec.apiSecurity?.enabled)
    this._setUnit(opts, 'appsec.apiSecurity.requestSampling', options.appsec.apiSecurity?.requestSampling)
```

File: packages/dd-trace/src/config.js [hunk 3]
```diff
@@ -810,11 +791,20 @@ class Config {
...
+    this._setValue(opts, 'appsec.blockedTemplateGraphql', maybeFile(options.appsec.blockedTemplateGraphql))
     this._setValue(opts, 'appsec.blockedTemplateHtml', maybeFile(options.appsec.blockedTemplateHtml))
     this._optsUnprocessed['appsec.blockedTemplateHtml'] = options.appsec.blockedTemplateHtml
     this._setValue(opts, 'appsec.blockedTemplateJson', maybeFile(options.appsec.blockedTemplateJson))
     this._optsUnprocessed['appsec.blockedTemplateJson'] = options.appsec.blockedTemplateJson
     this._setBoolean(opts, 'appsec.enabled', options.appsec.enabled)
+    let eventTracking = options.appsec.eventTracking && options.appsec.eventTracking.mode
```
Comment (simon-id) [inline packages/dd-trace/src/config.js:802]: missing optional chaining
```suggestion
    let eventTracking = options.appsec.eventTracking?.mode
```

File: packages/dd-trace/src/config.js (hunk 1 above)
Comment (simon-id) [reply packages/dd-trace/src/config.js]: @ida613 did you also change the mapping in the backend ?

**Review by Qard — DISMISSED** [review]

File: packages/dd-trace/src/config.js (hunk 1 above)
Comment (ida613) [reply packages/dd-trace/src/config.js]: yup

Comment (ida613) [reply packages/dd-trace/src/config.js]: https://github.com/DataDog/dd-go/pull/144559/files

Comment (simon-id) [reply packages/dd-trace/src/config.js]: I merged the above PR, approving this one :+1:

**Review by simon-id — APPROVED** [review]

**Comment by ida613** [comment]
this branch is messed up somehow, I'll start a new one

**Pull request closed without merging**